web/src/main/webapp/script.js

```diff
@@ -12,6 +12,48 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+function searchMe() {
+  var input, filter, ul, li, item, i, txtValue;
+  input = document.getElementById("myInput");
+  filter = input.value.toUpperCase();
+  ul = document.getElementById("task-list");
```

Comment:
ul and li are very unspecific variable names. In html, we would consider those "unordered list" and "list item" but since you are using javascript here, lets use more descriptive names. For example you can change "ul" to "taskList" and you can change "li" to "taskListItem". Also note that it is a bit odd they are called "tasks" when on the server you refer to them as "stocks".